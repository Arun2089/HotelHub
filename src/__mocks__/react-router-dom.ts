const mock = jest.requireActual("react-router-dom");

module.exports = {
  ...mock,
  useNavigate: jest.fn(),
  useParams: jest.fn(),
  useLocation: jest.fn(),
  useHistory: jest.fn(),
};
