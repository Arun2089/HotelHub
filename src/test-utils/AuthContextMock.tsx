import { AuthContext, AuthContextType } from "@src/context/AuthContext";

jest.mock("@src/utils/index.ts", () => ({
  fetchData: jest.fn().mockResolvedValue({
    id: 1,
    email: "johnDoe@gmail.com",
    role: { type: "admin" },
    username: "John Doe",
  }),
}));

type PropsWithChildren = {
  children: React.ReactNode;
  value: Partial<AuthContextType>;
};

const defaultValues = {
  user: {
    id: 1,
    email: "johnDoe@gmail.com",
    role: "admin",
    name: "John Doe",
  },
  updateUserDetails: jest.fn(),
  getUser: jest.fn().mockResolvedValue(true),
  isLoading: false,
  handleLogout: jest.fn(),
  handleLogin: jest.fn(),
  handleGoogleLogin: jest.fn(),
};

const AuthProviderMock = ({ children, value }: PropsWithChildren) => {
  return (
    <AuthContext.Provider value={{ ...defaultValues, ...value }}>
      {children}
    </AuthContext.Provider>
  );
};
export default AuthProviderMock;
