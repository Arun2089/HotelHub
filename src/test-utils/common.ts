export const setTimeOutMock = (ms: number) =>
  new Promise((resolve) => setTimeout(resolve, ms));
