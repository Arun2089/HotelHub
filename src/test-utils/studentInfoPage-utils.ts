import { fireEvent } from "@testing-library/react";

export const mockStudent = () => {
  return {
    studentName: "",
    roomNumber: null,
    mobileNumber: "",
    guardianName: "",
    guardianPhoneNumber: "",
    address: "",
    email: "",
    password: "",
    confirmPassword: "",
    govId: "",
    govIdImage: "",
  };
};

export const mockHandleChange = jest.fn();
export const mockValidateError = jest.fn();
export const mockGetHelperText = jest.fn();
export const mockHandleClick = jest.fn();

export function changeMultipleFields(
  textFields: HTMLElement[],
  details: { [key: string]: string | number | null | undefined }
) {
  textFields.forEach((textField) => {
    const name = textField.getAttribute("name");
    if (name) {
      fireEvent.change(textField, {
        target: { value: details[name] },
      });
    }
  });
}

export const studentDetails: {
  [key: string]: string | number | null | undefined;
} = {
  studentName: "John Doe",
  email: "john@gmail.com",
  mobileNumber: "9994567890",
};

export const guardianDetials: {
  [key: string]: string | number | null | undefined;
} = {
  guardianName: "John Doe",
  guardianPhoneNumber: "9994567890",
  address: "123, Main Street, New York",
};

export const passwordDetails: {
  [key: string]: string | number | null | undefined;
} = {
  password: "12345678",
  confirmPassword: "12345678",
};

export const govermentidDetails: {
  [key: string]: string | number | null | undefined;
} = {
  govId: "Pan Card",
  govIdImage: "Base64 string",
};
