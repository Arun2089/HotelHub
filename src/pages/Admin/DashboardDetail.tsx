import { useNavigate } from "react-router-dom";
import { Box, Skeleton, Paper, Typography } from "@mui/material";

import {
  AdminDashboardDataTypes,
  AdminDashboardDetailsTypes,
} from "@src/ts/types";

type DashboardPropsTypes = {
  dashboardData: AdminDashboardDataTypes | null;
  detail: AdminDashboardDetailsTypes;
};

const DashboardDetail = ({ dashboardData, detail }: DashboardPropsTypes) => {
  const naviate = useNavigate();
  const { label, field, icon: IconComponent, color, path } = detail;
  const clickableClass = path ? "cursor-pointer hover:brightness-90" : "";
  return (
    <Paper
      onClick={() => naviate(path)}
      key={label}
      className={`w-2/5 px-4 md:w-[48%] lg:w-1/4 h-24 lg:h-32 flex flex-grow md:shrink-0 lg:shrink justify-between items-center rounded-xl ${clickableClass} ${color} `}
    >
      {dashboardData ? (
        <>
          <IconComponent className="text-4xl md:text-5xl xl:text-7xl px-0 text-white" />
          <Box className="h-4/5 flex flex-col items-end justify-evenly">
            <Typography className="md:text-xl lg:text-2xl font-semibold  text-white">
              {label}
            </Typography>
            <Typography className=" text-2xl lg:text-3xl font-semibold text-white">
              {dashboardData[field as keyof AdminDashboardDataTypes]}
            </Typography>
          </Box>
        </>
      ) : (
        <>
          <Skeleton width="20%" height="50%" variant="rounded" />
          <Box className="w-3/5 h-full flex flex-col justify-center">
            <Skeleton width="90%" height="30%" />
            <Skeleton width="90%" height="30%" />
          </Box>
        </>
      )}
    </Paper>
  );
};

export default DashboardDetail;
