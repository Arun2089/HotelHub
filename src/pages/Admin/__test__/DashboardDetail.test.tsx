import { AccountBox } from "@mui/icons-material";
import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import DashboardDetail from "../DashboardDetail";
import { act } from "react-dom/test-utils";
import { useNavigate } from "react-router-dom";

const dashboardDataMock = {
  numberOfStudents: 50,
  complaitsPending: 10,
  numberOfRooms: 30,
  numberOfStaff: 15,
};
const detailMock = {
  label: "Students",
  icon: AccountBox,
  field: "numberOfStudents",
  color: "bg-primary-main",
  path: "students",
};

describe("DashboardDetail", () => {
  it("snapshot test", () => {
    const { asFragment } = render(
      <DashboardDetail dashboardData={dashboardDataMock} detail={detailMock} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders without error", () => {
    render(
      <DashboardDetail dashboardData={dashboardDataMock} detail={detailMock} />
    );
    expect(screen.getByText("Students")).toBeInTheDocument();
    expect(screen.getByText("50")).toBeInTheDocument();
  });

  it("renders skeleton when dashboardData is null", () => {
    render(<DashboardDetail dashboardData={null} detail={detailMock} />);
    const skeleton = document.querySelector(".MuiSkeleton-root");
    expect(skeleton).toBeInTheDocument();
    expect(screen.queryByText("50")).not.toBeInTheDocument();
  });

  it("onclick redirect to detail page", () => {
    const navigateMock = jest.fn();
    (useNavigate as jest.Mock).mockReturnValue(navigateMock);
    render(
      <DashboardDetail dashboardData={dashboardDataMock} detail={detailMock} />
    );
    const link = screen.getByText("50");
    act(() => {
      fireEvent.click(link);
    });
    expect(navigateMock).toHaveBeenCalled();
  });
});
