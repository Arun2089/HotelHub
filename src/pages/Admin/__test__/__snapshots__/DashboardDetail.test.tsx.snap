// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardDetail snapshot test 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 w-2/5 px-4 md:w-[48%] lg:w-1/4 h-24 lg:h-32 flex flex-grow md:shrink-0 lg:shrink justify-between items-center rounded-xl cursor-pointer hover:brightness-90 bg-primary-main  css-1ps6pg7-MuiPaper-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-4xl md:text-5xl xl:text-7xl px-0 text-white css-i4bv87-MuiSvgIcon-root"
      data-testid="AccountBoxIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 3c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm7 13H5v-.23c0-.62.28-1.2.76-1.58C7.47 15.82 9.64 15 12 15s4.53.82 6.24 2.19c.48.38.76.97.76 1.58V19z"
      />
    </svg>
    <div
      class="h-4/5 flex flex-col items-end justify-evenly MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 md:text-xl lg:text-2xl font-semibold  text-white css-ahj2mt-MuiTypography-root"
      >
        Students
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1  text-2xl lg:text-3xl font-semibold text-white css-ahj2mt-MuiTypography-root"
      >
        50
      </p>
    </div>
  </div>
</DocumentFragment>
`;
